File: VB/SpreadsheetDocServerChartAPISamples/CodeExamples/Charts1.vb
Imports DevExpress.Spreadsheet
Imports DevExpress.Spreadsheet.Charts
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace SpreadsheetChartAPISamples
    Public NotInheritable Class Charts

        Private Sub New()
        End Sub

        Private Shared Sub PieOfPieChart(ByVal workbook As Workbook)
            '            #Region "#PieOfPieChart"
            Dim worksheet As Worksheet = workbook.Worksheets("chartTask6")
            workbook.Worksheets.ActiveWorksheet = worksheet

            ' Create a Pie of Pie chart and specify its position.
            Dim chart As Chart = worksheet.Charts.Add(ChartType.PieOfPie, worksheet("B2:C11"))
            chart.TopLeftCell = worksheet.Cells("E2")
            chart.BottomRightCell = worksheet.Cells("L16")

            ' Specify the number of data points to be displayed in the secondary chart (the last four values).
            chart.Views(0).SplitType = OfPieSplitType.Position
            chart.Views(0).SplitPosition = 4

            ' Show data labels as percentage values.
            chart.Views(0).DataLabels.ShowPercent = True
            '            #End Region ' #PieOfPieChart
        End Sub
    End Class
End Namespace

File: CS/SpreadsheetDocServerChartAPISamples/CodeExamples/Charts1.cs
using DevExpress.Spreadsheet;
using DevExpress.Spreadsheet.Charts;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SpreadsheetChartAPISamples
{
    public static class Charts
    {
        static void PieOfPieChart(Workbook workbook)
        {
            #region #PieOfPieChart
            Worksheet worksheet = workbook.Worksheets["chartTask6"];
            workbook.Worksheets.ActiveWorksheet = worksheet;

            // Create a Pie of Pie chart and specify its position.
            Chart chart = worksheet.Charts.Add(ChartType.PieOfPie, worksheet["B2:C11"]);
            chart.TopLeftCell = worksheet.Cells["E2"];
            chart.BottomRightCell = worksheet.Cells["L16"];

            // Specify the number of data points to be displayed in the secondary chart (the last four values).
            chart.Views[0].SplitType = OfPieSplitType.Position;
            chart.Views[0].SplitPosition = 4;

            // Show data labels as percentage values.
            chart.Views[0].DataLabels.ShowPercent = true;
            #endregion #PieOfPieChart
        }
    }
}
